### PR TITLE
Book model testing merge

### PR DESCRIPTION
**Pull request by fcavazzoli**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -8,5 +8,6 @@ class User < ApplicationRecord
 
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  # Include default devise modules. Others available are:
```
Comment (floresv) [inline app/models/user.rb]: Why? 🤔

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,5 @@
 
 /log/*.log
 /log/*.keep
+log/
```
Comment (floresv) [inline .gitignore:3]: I think this work better than two before.

File: spec/factories/factory_bot.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  config.include FactoryBot::Syntax::Methods
+end
+
+FactoryBot.define do
+  factory :book do
+    genre { Faker::Book.genre }
+    author { Faker::Book.author }
+    image { Faker::File.file_name('foo/bar', 'baz', 'jpg') }
+    editorial { Faker::Book.publisher }
+    year { Faker::Date.birthday }
```
Comment (floresv) [inline spec/factories/factory_bot.rb]: Are you sure about this? I think maybe exists a better way with another [Faker::Date](https://github.com/stympy/faker/blob/master/doc/default/date.md)

**Review by floresv — CHANGES_REQUESTED** [review]
Please check my comments and resolve conflicts.

File: app/models/user.rb (hunk 1 above)
Comment (fcavazzoli) [reply app/models/user.rb]: i forgot to delete this comment

File: spec/factories/factory_bot.rb (hunk 3 above)
Comment (fcavazzoli) [reply spec/factories/factory_bot.rb]: i could use date.backward

File: log/development.log [hunk 4]
```diff
@@ -1,12 +1,7 @@
   [1m[35m (0.1ms)[0m  [1m[34mSELECT pg_try_advisory_lock(1580829076522303230)[0m
-  [1m[35m (0.8ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
-  [1m[36mActiveRecord::InternalMetadata Load (0.6ms)[0m  [1m[34mSELECT  "ar_internal_metadata".* FROM "ar_internal_metadata" WHERE "ar_internal_metadata"."key" = $1 LIMIT $2[0m  [["key", "environment"], ["LIMIT", 1]]
+  [1m[35m (0.3ms)[0m  [1m[34mSELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC[0m
```
Comment (floresv) [inline log/development.log]: remove this file. 🙏

File: spec/factories/factory_bot.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'factory_bot'
+require 'faker'
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
```
Comment (floresv) [inline spec/factories/factory_bot.rb]: Do you need all these 5 lines before here? 🤔 Review it, please.

File: Gemfile [hunk 6]
```diff
@@ -55,6 +55,10 @@ group :development, :test do
   # Adds support for Capybara system testing and selenium driver
   gem 'capybara', '>= 2.15'
   gem 'selenium-webdriver'
+  gem 'database_cleaner'
```
Comment (floresv) [inline Gemfile:58]: Do you config database cleaner?

File: spec/factories/factory_bot.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+FactoryBot.define do
+  factory :book do
+    genre { Faker::Book.genre }
+    author { Faker::Book.author }
+    image { Faker::File.file_name('foo/bar', 'baz', 'jpg') }
+    editorial { Faker::Book.publisher }
+    year { Faker::Date.backward }
```
Comment (floresv) [inline spec/factories/factory_bot.rb]: year field its string, but backward generate a Date, no? Review it please 😉

Comment (fcavazzoli) [reply spec/factories/factory_bot.rb]: shoud i  keep the string type ? date type sounds better

File: Gemfile (hunk 6 above)
Comment (fcavazzoli) [reply Gemfile:58]: i didnt, shoul i? or just remove the gem  ?

File: spec/factories/factory_bot.rb (hunk 5 above)
Comment (fcavazzoli) [reply spec/factories/factory_bot.rb]: removed, this was a desperate try when it wasnt working hahah

File: Gemfile (hunk 6 above)
Comment (fcavazzoli) [reply Gemfile:58]: configured.

File: spec/factories/factory_bot.rb (hunk 7 above)
Comment (fcavazzoli) [reply spec/factories/factory_bot.rb]: i took the string option using .year.to_s methods

File: spec/rails_helper.rb [hunk 8]
```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'spec_helper'
+require 'factories/factory_bot.rb'
```
Comment (floresv) [inline spec/rails_helper.rb]: Why?

File: spec/models/book_spec.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
```
Comment (floresv) [inline spec/models/book_spec.rb:3]: Do you really need this?

File: spec/factories/factory_bot.rb [hunk 10]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require 'factory_bot'
+
+FactoryBot.define do
```
Comment (floresv) [inline spec/factories/factory_bot.rb]: A factory must be in line with her filename. Read [this](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#rspec) please. Remember, how [factories](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#defining-factories) works.

Comment (floresv) [reply spec/factories/factory_bot.rb]: Also, read [this](https://semaphoreci.com/community/tutorials/working-effectively-with-data-factories-using-factorygirl).

File: spec/rails_helper.rb (hunk 8 above)
Comment (fcavazzoli) [reply spec/rails_helper.rb]: tue, this was unnecesary

File: spec/models/book_spec.rb (hunk 9 above)
Comment (fcavazzoli) [reply spec/models/book_spec.rb:3]: i do, without this line it doesnt work :/

File: spec/factories/factory_bot.rb (hunk 10 above)
Comment (fcavazzoli) [reply spec/factories/factory_bot.rb]: i did refactor this part of the code :)

**Review by floresv — APPROVED** [review]
LGTM

**Pull request merged**